### PR TITLE
Closes #3782:  flip function to match numpy

### LINKED ISSUE #3782: flip function to match numpy
Add a `flip` function to match numpy:
https://numpy.org/doc/stable/reference/generated/numpy.flip.html

There is already a `flip` function in the array api, so that may be a good starting point.

### PR DESCRIPTION
**Pull request by ajpotts**
This PR adds a flip function to match numpy:
https://numpy.org/doc/2.1/reference/generated/numpy.flip.html

Closes #3782: flip function to match numpy

### CODE REVIEW

**Comment by drculhane** [comment]
When I download this and run pytest on test_manipulation_functions, test_flip_string fails in single dim, and both test_flip_string and test_flip_multi_dim_bool fail in multi dim.  But I see that it passed all the checks, so I'm not sure what to make of this.

image

**Comment by ajpotts** [comment]
Fixed two problems likely causing it to fail locally but not in the CI:

- The tests had not been added to `pytest.ini`
- `test_flip_multi_dim_bool` had poor logic that needed to be replace.

Thanks Andy, for pointing this out. @drculhane

**Comment by drculhane** [comment]
When I download this and run pytest on test_manipulation_functions, test_flip_string fails in single dim, and both test_flip_string and test_flip_multi_dim_bool fail in multi dim.  But I see that it passed all the checks, so I'm not sure what to make of this.

image

**Comment by drculhane** [comment]
This is what I get.  I had it sent to a text file, since these sorts of things are always long.

[errors.txt](https://github.com/user-attachments/files/17248992/errors.txt)

**Review by drculhane — APPROVED** [review]
Looks good.

**Review by jaketrookman — APPROVED** [review]
Looks good

File: tests/numpy/manipulation_functions_test.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import pytest
+
+import arkouda as ak
+from arkouda.categorical import Categorical
+from arkouda.testing import assert_equal
+
+seed = pytest.seed
+
+
+class TestJoin:
```
Comment (stress-tess) [inline tests/numpy/manipulation_functions_test.py:11]: Im guessing this wasn't the intended class name

Comment (stress-tess) [inline tests/numpy/manipulation_functions_test.py:11]: Im guessing this wasn't the intended class name. unfortunately it seems `TestManipulationFunctions` is already taken

**Review by stress-tess — APPROVED** [review]
looks good to me! small comment on the test class name

File: tests/numpy/manipulation_functions_test.py (hunk 1 above)
Comment (ajpotts) [reply tests/numpy/manipulation_functions_test.py]: I believe that was a copy-and-paste error.  Thanks for catching it!  It should be fixed now.

**Review by stress-tess — APPROVED** [review]

**Pull request merged**